crs/v1alpha2/controlplane_ubuntu.yaml

```diff
@@ -42,22 +42,29 @@ spec:
     - mv /tmp/akeys /home/ubuntu/.ssh/authorized_keys
     - chown ubuntu:ubuntu /home/ubuntu/.ssh/authorized_keys
     - apt update -y
+    - netplan apply
     - >-
-      apt install net-tools gcc linux-headers-$(uname -r) keepalived
+      apt install net-tools gcc linux-headers-$(uname -r) bridge-utils
       apt-transport-https ca-certificates curl gnupg-agent
       software-properties-common -y
+    - apt install -y keepalived && systemctl stop keepalived
     - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
     - add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
     - curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
     - echo 'deb https://apt.kubernetes.io/ kubernetes-xenial main' > /etc/apt/sources.list.d/kubernetes.list
     - apt update -y
     - apt install docker-ce docker-ce-cli containerd.io kubelet kubeadm kubectl -y
-    - systemctl enable --now docker keepalived kubelet
+    - systemctl enable --now docker kubelet
+    - if (curl -sk --max-time 10 https://${CLUSTER_APIENDPOINT_IP}:6443/healthz); then echo "keepalived already running";else systemctl start keepalived; fi
     - usermod -aG docker ubuntu
   postKubeadmCommands:
     - mkdir -p /home/ubuntu/.kube
     - cp /etc/kubernetes/admin.conf /home/ubuntu/.kube/config
+    - systemctl enable --now keepalived
     - chown ubuntu:ubuntu /home/ubuntu/.kube/config
+    - curl https://docs.projectcalico.org/v3.10/manifests/calico.yaml -O
+    - sed -i -e "s?192.168.0.0/16?$POD_CIDR?g" calico.yaml
+    - kubectl --kubeconfig /home/ubuntu/.kube/config apply -f calico.yaml
```

Comment:
should we install calico here ? from an upstream PR, it was discussed and kinda agreed to not deploy any SDN, that is an addon

Reply:
I see no harm in it since it is for testing mainly ...

Reply:
I have removed them anyhow since you mentioned it is agreed not to deploy any SDN